seatunnel-connectors-v2/connector-rabbitmq/src/main/java/org/apache/seatunnel/connectors/seatunnel/rabbitmq/config/RabbitmqBaseOptions.java

```diff
@@ -0,0 +1,185 @@
+                    .withDescription("if true, enables topology recovery");
+
+    public static final Option<Integer> CONNECTION_TIMEOUT =
+            Options.key("connection_timeout")
+                    .intType()
+                    .noDefaultValue()
+                    .withDescription("connection TCP establishment timeout in milliseconds");
+
+    public static final Option<String> ROUTING_KEY =
+            Options.key("routing_key")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription("the routing key to publish the message to");
+
+    public static final Option<String> EXCHANGE =
+            Options.key("exchange")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription("the exchange to publish the message to");
+
+    public static final Option<Boolean> FOR_E2E_TESTING =
+            Options.key("for_e2e_testing")
+                    .booleanType()
+                    .noDefaultValue()
+                    .withDescription("use to recognize E2E mode");
+
+    public static final Option<Map<String, String>> RABBITMQ_CONFIG =
+            Options.key("rabbitmq.config")
+                    .mapType()
+                    .defaultValue(Collections.emptyMap())
+                    .withDescription(
+                            "In addition to the above parameters that must be specified by the RabbitMQ client, the user can also specify multiple non-mandatory parameters for the client, "
+                                    + "covering [all the parameters specified in the official RabbitMQ document](https://www.rabbitmq.com/configure.html).");
+
+    public static final Option<Boolean> USE_CORRELATION_ID =
+            Options.key("use_correlation_id")
+                    .booleanType()
+                    .noDefaultValue()
+                    .withDescription(
+                            "Whether the messages received are supplied with a unique"
+                                    + " id to deduplicate messages (in case of failed acknowledgments).");
+
+    public static final Option<Boolean> DURABLE =
+            Options.key("durable")
+                    .booleanType()
+                    .defaultValue(true)
+                    .withDescription(
+                            "true: The queue will survive a server restart."
+                                    + " false: The queue will be deleted on server restart.");
+
+    public static final Option<Boolean> EXCLUSIVE =
+            Options.key("exclusive")
+                    .booleanType()
+                    .defaultValue(false)
+                    .withDescription(
+                            "true: The queue is used only by the current connection and will be deleted when the connection closes."
+                                    + " false: The queue can be used by multiple connections.");
+
+    public static final Option<Boolean> AUTO_DELETE =
```

Comment:
All parameter in `Option` should describe in the Factory `optionRule` method. 
Is it required, optional. Bind with other parameter etc.